src/op_analytics/cli/subcommands/pulls/defillama.py

```diff
@@ -22,50 +24,70 @@
 METADATA_TABLE = "defillama_stablecoins_metadata"
 
 
-def get_data(session, url):
-    start = time.time()
-    resp = session.request(
-        method="GET",
-        url=url,
-        headers={"Content-Type": "application/json"},
-    ).json()
-    log.info(f"Fetched from {url}: {time.time() - start:.2f} seconds")
-    return resp
-
-
-def process_breakdown_stables(data):
-    peg_type = data["pegType"]
-    balances = data["chainBalances"]
+def get_data(session, url: str) -> Optional[dict]:
+    """
+    Fetches data from a given URL using the provided session.
 
-    rows = []
-    for chain, balance in balances.items():
-        tokens = balance["tokens"]
+    Args:
+        session: The HTTP session for making requests.
+        url (str): The URL to fetch data from.
 
-        for datapoint in tokens:
-            row = {}
-            row["chain"] = chain
-            row["dt"] = dt_fromepoch(datapoint["date"])
+    Returns:
+        Optional[dict]: The JSON response as a dictionary, or None if an error occurred.
+    """
+    start = time.time()
+    try:
+        response = session.get(url, headers={"Content-Type": "application/json"})
+        response.raise_for_status()
+        resp_json = response.json()
+        log.info(f"Fetched from {url}: {time.time() - start:.2f} seconds")
+        return resp_json
+    except Exception as e:
+        log.error(f"Failed to fetch data from {url}: {e}")
+        return None
```

Comment:
My recommendation would be not to catch the exception. If there is an error we want to be aware of it.  But am I missing something?

While working on the github analytics api pull I also realized my get_data method was not checking for the http status code of the response. So perhaps we should also check for status = 200 and raise an exception if we don't get 200. 

Since we are mostly doing the same for all http requests we  can move the get_data() function over to `op_coreutils.request` and not redefine the same functionality in `defillama.py` and `l2beat.py` and `github_analytics.py`.  

Let me take an initial stab at the get_data function in the github_anlytics pr (#955 ) and we can then go back and use that function in this one.

Reply:
here is the common version of the data that I think we could use here: https://github.com/ethereum-optimism/op-analytics/blob/main/packages/op-coreutils/src/op_coreutils/request.py#L28